src/Framework/TestCase.php

```diff
@@ -1535,6 +1564,48 @@ protected function onNotSuccessfulTest(Throwable $t): never
         throw $t;
     }
 
+    /**
+     * @throws ExpectationFailedException
+     */
+    private function verifyDeprecationExpectations(): void
+    {
+        foreach ($this->expectedUserDeprecationMessage as $deprecationExpectation) {
+            $this->numberOfAssertionsPerformed++;
+
+            if (!in_array($deprecationExpectation, DeprecationCollector::deprecations(), true)) {
```

Comment:
This requires an exact match on the deprecation message. For `expectExceptionMessage`, PHPUnit checks it with `IsOrContains` semantic. Shouldn't it be consistent here as well ?

Reply:
I prefer this inconsistency over repeating the mistake that was made with `expectExceptionMessage()`.

Reply:
If this is considered a mistake, then OK.

Maybe `expectExceptionMessage` should then mark a test as risky when it matches with Contains semantic instead of Is semantic (not sure this is doable in the current architecture)

Reply:
Yes, I want to deprecate sub-string matching in `expectExceptionMessage()` at some point with the goal removing support for it entirely later.

Reply:
Maybe you should add a test enforcing that `expectedUserDeprecationMessage('message')` will *not* match `expectedUserDeprecationMessage('another message')` in the tests of the feature. AFAICT, implementing an IsOrContains semantic in the check would still keep the current tests green.